tern/analyze/default/default_common.py

```diff
@@ -150,3 +153,21 @@ def abort_analysis():
     """Abort due to some external event"""
     rootfs.recover()
     sys.exit(1)
+
+
+def default_common_procedure(image, redo, driver, extension, args=None, image_file=False):
+    """Setup the image object and anything else for analysis"""
+    # Add a Notice object for each layer
+    for layer in image.layers:
+        origin_str = 'Layer {}'.format(layer.layer_index)
+        layer.origins.add_notice_origin(origin_str)
+    # Set up working directories and mount points
+    rootfs.set_up()
+    # analyze the image
+    cimage.analyze(image, redo, driver, extension)
```

Comment:
@nishakm I need cimage here in 167th line, that's why I'm importing it in default_common. One way around is that I remove this entire function from `default_common.py` to `common.py`, this will resolve the circular imports issue that you mentioned.